mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearchIndex.java

```diff
@@ -55,6 +58,11 @@
 
     EnumSet<MailboxManager.SearchCapabilities> getSupportedCapabilities(EnumSet<MailboxManager.MessageCapabilities> messageCapabilities);
 
+    /**
+     * Retrieves flags of an indexed message
+     */
+    Mono<Flags> retrieveIndexedFlags(MailboxSession session, MailboxId mailboxId, MessageUid uid) throws MailboxException;
```

Comment:
Retrieving indexed data likely don't make search for the scanning implementation, it should be defined in `ListeningMessageSearchIndex`

Also I would make session parameter the last to keep consistent with the implicit mailbox-api convention.